spaceMissionManager-service/src/main/java/cz/muni/fi/services/impl/MissionServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+                        ", name='" + mission.getName() + '\'' +
+                        ", destination='" + mission.getDestination() + '\'' +
+                        ", eta=" + mission.getEta() +
+                        ", missionDescription='" + mission.getMissionDescription() + '\'' +
+                        ", endDate=" + mission.getEndDate() +
+                    '}');
+        try {
+            missionDao.updateMission(mission);
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not archive mission.", e);
+        }
+    }
+
+    @Override
+    public void createMission(Mission mission) throws DataAccessException {
+        if (mission == null){
+            throw new IllegalArgumentException("Mission must not be null.");
+        }
+        try {
+            missionDao.createMission(mission);
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not create mission.", e);
+        }
+    }
+
+    @Override
+    public void cancelMission(Mission mission) throws DataAccessException {
+        if (mission == null){
+            throw new IllegalArgumentException("Mission must not be null.");
+        }
+        try {
+            missionDao.cancelMission(mission);
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not cancel mission.", e);
+        }
+    }
+
+    @Override
+    public List<Mission> findAllMissions() throws DataAccessException {
+        try {
+            return Collections.unmodifiableList(missionDao.findAllMissions());
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not find missions.", e);
+        }
+    }
+
+    @Override
+    public List<Mission> findAllMissions(boolean active) throws DataAccessException {
+        try {
+            return Collections.unmodifiableList(missionDao.findAllMissions(active));
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not find missions.", e);
+        }
+    }
+
+    @Override
+    public Mission findMissionById(Long id) throws DataAccessException {
+        if (id == null){
+            throw new IllegalArgumentException("Id must not be null.");
+        }
+        try {
+            return missionDao.findMissionById(id);
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not find mission.", e);
+        }
+    }
+
+    @Override
+    public void updateMission(Mission mission) throws DataAccessException {
+        if (mission == null){
+            throw new IllegalArgumentException("Mission must not be null.");
+        }
+        if (mission.getId() == null){
+            throw new IllegalArgumentException("Mission ID is null, could not update.");
+        }
+        boolean missionActive;
+        try {
+            missionActive = mission.isActive() && missionDao.findMissionById(mission.getId()).isActive();
+        } catch (Throwable e){
```

Comment:
You can't catch everything when you are just searching for mission in DB.
Why don't you just check if that mission which you get from DB is not null? Then you don't have to catch anything here.